packages/office-addin-manifest/src/validate.ts

```diff
@@ -65,13 +66,24 @@ export async function validateManifest(
   manifestPath: string,
   verifyProduction: boolean = false
 ): Promise<ManifestValidation> {
-  try {
-    const validation: ManifestValidation = new ManifestValidation();
-    const clientId: string = verifyProduction ? "Default" : "devx";
-
-    // read the manifest file to ensure the file path is valid
-    await OfficeAddinManifest.readManifestFile(manifestPath);
-
+  const validation: ManifestValidation = new ManifestValidation();
+  const clientId: string = verifyProduction ? "Default" : "devx";
+
+  // read the manifest file to ensure the file path is valid
+  await OfficeAddinManifest.readManifestFile(manifestPath);
+
+  if (manifestPath.endsWith(".json")) {
+    const manifest: TeamsAppManifest = await ManifestUtil.loadFromPath(manifestPath);
+    const validationResult: string[] = await ManifestUtil.validateManifest(manifest);
+
+    if (validationResult.length !== 0) {
+      // There are errors
+      validation.isValid = false;
+      validation.jsonErrors = validationResult;
```

Comment:
It may be quicker to just add a json manifest version of the results, but I don't think it's good design to have one off properties like this.  We aren't going to have json and xml results at the same time.

Other things to consider . . . the json manifest results output should looks as similar to the xaml ones (as much as possible) and should allow for warnings vs. error (I'm not sure if there are warnings in the json manifest validation results . . . but you should check).

Reply:
The json validation is literally just an array of strings. Each string is an error, they just have errors, no other case.
But I agree they should be more similar, so I adapted this new validation format to ours.